python/lsst/ci/build.py

```diff
@@ -345,6 +393,144 @@ def run(args):
 
         b = Builder(build_dir, manifest, progress, eups_obj)
         b.rm_status()
-        retcode = b.build()
-        b.write_status()
-        sys.exit(retcode == 0)
+
+        try:
+            if agent == "unknown_agent":
+                raise UnknownAgentError(agent)
```

Comment:
Is it better to wait until just about to build before raising this?  I'd think it would be better to raise as soon as recognized (in `get_agent()`) or else just print the warning and let the build go without having an exception at all.